File: back-end/GeekSpot.API/Models/Usuario.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace GeekSpot.API.Models
{
    public class Usuario
    {
        [Key]
        public int UsuarioId { get; set; }
        public string? NomeCompleto { get; set; }
        public string? Email { get; set; }
        public string? NomeUsuarioSistema { get; set; }
        public string? Senha { get; set; }

        // Fk (De lá pra cá);
        public int UsuarioTipoId { get; set; }
        public UsuarioTipo? UsuariosTipos { get; set; }

        public string? Foto { get; set; }
        public DateTime DataRegistro { get; set; }
        public DateTime DataOnline { get; set; }
        public int IsAtivo { get; set; }
        public int IsPremium { get; set; }
        public int IsVerificado { get; set; }

        // Fk (De cá pra lá);
        [JsonIgnore]
        public UsuarioInformacao? UsuariosInformacoes { get; set; }
    }
}

File: front-end/utils/types/usuario.ts
import { StaticImageData } from 'next/image';
import iUsuarioInformacoes from './usuarioInformacoes';
import iUsuarioTipo from './usuarioTipo';

export default interface iUsuario {
    usuarioId: null;
    nomeCompleto: string;
    email: string;
    nomeUsuarioSistema: string;
    token: string | null;
    usuarioTipoId: number;
    usuariosTipos: iUsuarioTipo;
    foto: string | null;
    dataRegistro: Date | null;
    dataOnline: Date | null;
    isAtivo: boolean;
    isPremium: boolean | null;
    isVerificado: boolean | null;
    isDesativado: boolean | null;
    usuariosInformacoes: iUsuarioInformacoes;
    erro: boolean;
    codigoErro: number | null;
    mensagemErro: string | null;

    // Propriedades extra que não estão na entidade Usuario que são usadas no front-end apenas;
    genero: string | null;
    fotoPerfilAlternativa: StaticImageData | null;
    cep: string | null;
    isEmailVerificacaoContaEnviado: boolean | null;
}

